rootfs/scheduler/resources/pod.py

```diff
@@ -384,14 +384,20 @@ def _set_image_secret(self, data, namespace, **kwargs):
         data['imagePullSecrets'] = [{'name': secret_name}]
 
     def delete(self, namespace, name):
+        # get timeout info from pod
+        pod = self.pod.get(namespace, name).json()
```

Comment:
is there a chance of this giving 404 or do we want to handle that case in the function where it is called?